data/transactions/logic/crypto_test.go

```diff
@@ -117,6 +117,76 @@ byte 0x98D2C31612EA500279B6753E5F6E780CA63EBA8274049664DAD66A2565ED1D2A
 	testAccepts(t, progText, 1)
 }
 
+func TestMimc(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	// Test vectors from https://github.com/giuliop/test-mimc-opcodes/blob/main/mimctest/main.go
+	// generated by instantiating a zk-circuit that processes MiMC hash of the preimages.
+	// We test success for 32-byte and 96-byte preimages, and failure for preimage input size of 0,
+	// input size not multiple of 32 bytes, and chunks representing values greater than the modulus.
```

Comment:
Is there no more direct test vector possible, say extracted within gnark repo? Seems weird to have tests that depend on some other code, not in repo, being correct.

Reply:
I like those tests because the only reason to use the mimc opcode is to match its use in a zk-circuit.
We really want them to match.

Reply:
I commented how we generated the test cases as discussed